src/Ipac_Table_Writer/internals.cxx

```diff
@@ -108,94 +192,257 @@ void tablator::Ipac_Table_Writer::write_selected_records(
+                                                    size_t col_width,
+                                                    size_t effective_array_size) {
+    auto& column = table.columns[col_id];
+    auto& null_value = column.field_properties.values.null;
+    const std::string& null_str =
+            (null_value.empty()) ? tablator::Table::DEFAULT_NULL_VALUE : null_value;
+
+    for (size_t element = 0; element < effective_array_size; ++element) {
+        os << std::setw(col_width) << null_str << "|";
+    }
+}
+
+/**********************************************************/
+
+void tablator::Ipac_Table_Writer::write_single_value(
+        const tablator::Table& table, std::ostream& os, size_t column_id,
+        size_t curr_row_offset,
+        const std::vector<tablator::Data_Type>& datatypes_for_writing) {
+    auto& column = table.columns[column_id];
+    auto width = table.get_column_widths()[column_id];
+    tablator::Data_Type active_datatype = datatypes_for_writing[column_id];
+    if (table.is_null(curr_row_offset, column_id)) {
+        auto& null_value = column.field_properties.values.null;
+        const std::string& null_str =
+                (null_value.empty()) ? tablator::Table::DEFAULT_NULL_VALUE : null_value;
         size_t effective_array_size =
-                get_effective_array_size(datatypes_for_writing[i], column.array_size);
-        auto null = column.field_properties.values.null;
-        std::string null_str = (null.empty()) ? DEFAULT_NULL_VALUE : null;
+                get_effective_array_size(active_datatype, column.array_size);
 
         for (size_t element = 0; element < effective_array_size; ++element) {
-            os << std::setw(ipac_column_widths[i]) << null_str << "|";
+            os << IPAC_COLUMN_SEPARATOR << std::setw(width);
+            os << null_str;
         }
+    } else if (active_datatype == tablator::Data_Type::UINT8_LE) {
+        // Do this case manually because write_type_as_ascii()
+        // isn't equipped to write bytes as ints, as IPAC_FORMAT
+        // requires.
+        size_t base_offset = curr_row_offset + table.offsets[column_id];
+        uint8_t const* curr_data = table.data.data() + base_offset;
+        size_t element_size = data_size(active_datatype);
+
+        for (size_t element = 0; element < column.array_size; ++element) {
+            os << IPAC_COLUMN_SEPARATOR << std::setw(width);
+            os << static_cast<const uint16_t>(*curr_data);
+            curr_data += element_size;
+        }
+    } else {
+        size_t base_offset = curr_row_offset + table.offsets[column_id];
+        uint8_t const* curr_data = table.data.data() + base_offset;
+
+        os << IPAC_COLUMN_SEPARATOR << std::setw(width);
+        std::stringstream ss_temp;
+        write_type_as_ascii(ss_temp, column.type, column.array_size, curr_data, width);
+        std::string s(ss_temp.str());
+        os << boost::replace_all_copy(s, "\n", " ");
```

Comment:
I don't know how performance critical this function is, but if it is, I would avoid the copy and mutate `s` instead. Or even better, `write_type_as_ascii` could write to `os` directly.